### PR TITLE
[CPU] General denormals optimization

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
computation with denormals is very time consuming. Flushing denormals to zero could significantly improve the perf while not be comply with IEEE standard, in most case this behavior have little impact for model accuracy. Users could enable this opt if see none or little impact on accuracy.
 - *Introducing Denormals Optimization property in IE core. Use core.SetConfig({{CONFIG_KEY(DENORMALS_OPTIMIZATION), "YES"}}) or include into config file to specify*
 - *CPU plugin optimize it by setting FTZ and DAZ in mxcsr explicitly when it's feasible to improve denormals computation perf when the property is set*
 - *benchmark_app update to include this option*

### Tickets:
 - *85654*

### CODE REVIEW

File: tools/benchmark_tool/openvino/tools/benchmark/parameters.py [hunk 1]
```diff
@@ -152,6 +152,8 @@ def parse_args():
     args.add_argument('-imean', '--input_mean', type=str, required=False, default='',
                       help="Optional. Mean values to be used for the input image per channel.\n Values to be provided in the [R, G, B] format. Can be defined for desired input of the model.\n"
                            "Example: -imean data[255,255,255],info[255,255,255]\n")
+    args.add_argument('-do', '--denormals_optimization', type=str2bool, required=False, default=False, nargs='?', const=True,
```
Comment (jiwaszki) [inline tools/benchmark_tool/openvino/tools/benchmark/parameters.py:155]:
```suggestion
    args.add_argument('-dopt', '--denormals_optimization', type=str2bool, required=False, default=False, nargs='?', const=True,
```
I would call it `dopt` in all places. It is more meaningful in my opinion.

Comment (chenhu-wang) [reply tools/benchmark_tool/openvino/tools/benchmark/parameters.py:155]: @jiwaszki ok, agree. will update!

File: src/plugins/intel_cpu/src/plugin.cpp [hunk 2]
```diff
@@ -725,6 +726,24 @@ Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork &network, const std
...
+        //     flush_to_zero(true);
+        //     denormals_as_zero(true);
+        // } else if (conf.denormalsOptMode == Config::DenormalsOptMode::DO_Off) {
+        //     std::cout << "DenormalsOptMode is set to OFF." << std::endl;
+        //     flush_to_zero(false);
+        //     denormals_as_zero(false);
+        // }
+        // std::cout << "DenormalsOptMode is not explicitly set and keep." << std::endl;
+    }
+
```
Comment (usstq) [inline src/plugins/intel_cpu/src/plugin.cpp:746]: The default behavior here is important, considering user may set FTZ DAZ outside OpenVINO and he/she maybe surprised by our default setting on these flags, maybe by default we should do nothing, but in benchmark_app we can enable this feature by default

Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/plugin.cpp:746]: This is actually already how it works now. The default is [DO_KEEP](https://github.com/openvinotoolkit/openvino/pull/11883/files#diff-76c2faa205c3624b84079d81a130db2ddac8513dd1c7df85ea327b51171793beR51). If user do not set this property explicitly, the DO behavior inherit from parent environment where user may set outside OV. Only the value of key is ["YES" or "NO"](https://github.com/openvinotoolkit/openvino/pull/11883/files#diff-bfc4c8727c69c70f65b7b1fb1261169bc09c9b202865cd8048ab892999ba0717R151), we set this accordingly.

Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/plugin.cpp:746]: The code here is hard-code for DLB test temporary. will revert a later time

File: src/plugins/intel_cpu/src/utils/denormals.hpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            }
+        }
+    #else
+        bool flush_to_zero(bool on) { return false; }
+        bool denormals_as_zero(bool on) { return false; }
+    #endif
+#endif
+
+}   // namespace intel_cpu
+}   // namespace ov
```
Comment (usstq) [inline src/plugins/intel_cpu/src/utils/denormals.hpp:115]: I saw [many places][1][2] suggest setting FTZ and DAZ with simple intrinsics like :
```cpp
#include <xmmintrin.h>
#include <pmmintrin.h>

int main()
{
	_MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);         // FTZ
	_MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON); // DAZ
	return 0;
}
```

I personally check it on VisualStudio2019 and GCC and it works, should we switch to this simpler method?

[1]: https://www.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/compiler-reference/floating-point-operations/set-the-ftz-and-daz-flags.html
[2]: https://oneapi-src.github.io/oneDNN/dev_guide_data_types.html#floating-point-environment

File: src/plugins/intel_cpu/src/plugin.cpp (hunk 2 above)
Comment (usstq) [reply src/plugins/intel_cpu/src/plugin.cpp:746]: Another concern is, the setup of FTZ DAZ will not take effect immediately until Loadnetwork is called, which is pretty weird from user's perspective, especially when user want to turn of these flags

File: src/plugins/intel_cpu/src/utils/denormals.hpp (hunk 3 above)
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/utils/denormals.hpp:115]: We need cross ISA/OS/compiler solution, the simple one is not safe.

**Comment by chenhu-wang** [comment]
@kblaszczak-intel , Hi could you help review .md related changes?

**Review by usstq — APPROVED** [review]

File: src/inference/include/openvino/runtime/properties.hpp [hunk 4]
```diff
@@ -457,6 +457,12 @@ static constexpr Property<Level> level{"LOG_LEVEL"};
  */
 static constexpr Property<std::string> cache_dir{"CACHE_DIR"};
 
+/**
+ * @brief The name for setting how to handle denormals.
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/properties.hpp:461]: can we have extensive description as for API 1.0?

Comment (wangleis) [reply src/inference/include/openvino/runtime/properties.hpp:461]: @chenhu-wang Could you create one more PR for description?

Comment (chenhu-wang) [reply src/inference/include/openvino/runtime/properties.hpp:461]: @ilya-lavrenov @wangleis , the extensive description is done along with Doc PR [12132](https://github.com/openvinotoolkit/openvino/pull/12132) [12127](https://github.com/openvinotoolkit/openvino/pull/12127)

**Pull request merged**